backend/code/m-booking/src/test/java/com/mbooking/service/AuthenticationServiceIntegrationTest.java

```diff
@@ -0,0 +1,167 @@
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.test.annotation.Rollback;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.transaction.annotation.Transactional;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ActiveProfiles("test_h2")
+public class AuthenticationServiceIntegrationTest {
+
+    @Autowired
+    private AuthenticationService authenticationService;
+
+    @Autowired
+    private CustomerRepository customerRepository;
+
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+    @Test(expected = ApiAuthException.class)
+    public void when_LoginCredentialsInvalid() {
+        String email = "email@email.com";
+        String password = "12412412";
+        authenticationService.login(email, password);
+    }
+
+    @Test(expected = ApiAuthException.class)
+    @Transactional
+    @Rollback
+    public void when_LoginCredentialsValid_And_EmailNotConfirmed() {
+        String email = "email@email.com";
+        String password = "12345";
+        String encodedPassword = passwordEncoder.encode(password);
+        Customer customer = new Customer();
+        customer.setFirstname("");
+        customer.setLastname("");
+        customer.setEmail(email);
+        customer.setPassword(encodedPassword);
+        customer.setEmailConfirmed(false);
+        customer.setBanned(false);
+        customerRepository.save(customer);
+
+        try {
+            authenticationService.login(email, password);
+        } catch (ApiAuthException ex) {
+            Assert.assertEquals("Please confirm registration!", ex.getMessage());
+            throw ex;
+        }
+    }
+
+    @Test(expected = ApiAuthException.class)
+    @Transactional
+    @Rollback
+    public void when_LoginCredentialsValid_And_UserIsBanned() {
+        String email = "email@email.com";
+        String password = "12345";
+        String encodedPassword = passwordEncoder.encode(password);
+        Customer customer = new Customer();
+        customer.setFirstname("");
+        customer.setLastname("");
+        customer.setEmail(email);
+        customer.setPassword(encodedPassword);
+        customer.setEmailConfirmed(true);
+        customer.setBanned(true);
+        customerRepository.save(customer);
+
+        try {
+            authenticationService.login(email, password);
+        } catch (ApiAuthException ex) {
+            Assert.assertEquals("Your account is banned!", ex.getMessage());
+            throw ex;
+        }
+    }
+
+    @Test
+    @Transactional
+    @Rollback
+    public void when_LoginCredentialsValid() {
```

Comment:
It would be nice to add what happens in test name -> whenLoginCredentialsValid_thenSaveUser ;)